quota/metrics.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package quota
+
+import "github.com/google/trillian/monitoring"
+
+// Metrics groups all quota-related metrics.
+// The metrics represented here are not meant to be maintained by the quota subsystem
+// implementation.  Instead, they're meant to be updated by the quota's callers, in order to record
```

Comment:
Could this be achieved by an indirection layer? A pass-through QuotaManager shim which handles the metrics and then calls on to the actual registered quota manager impl?